addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/converter/swagger/RequestModelConverter.java

```diff
@@ -42,6 +44,17 @@ public RequestModel convert(OperationModel operationModel, Generators generators
         requestModel.setBody(this.generateBody());
         requestModel.setMethod(operationModel.getRequestMethod());
         requestModel.setHeaders(this.generateHeaders());
+        List<HttpHeaderField> headers = requestModel.getHeaders();
+        for (HttpHeaderField hhf : headers) {
+            if (hhf.getValue().contains("multipart/form-data")) {
+                String requestBody = requestModel.getBody();
+                String boundary = requestBody.substring(2, 38);
+                HttpHeaderField boundaryHeader =
+                        new HttpHeaderField(
+                                hhf.getName(), hhf.getValue() + "; boundary=" + boundary);
+                headers.set(headers.indexOf(hhf), boundaryHeader);
+            }
+        }
```

Comment:
Preferable to do this when generating the headers, the body was already generated so it can make use of it.